### PR TITLE
feat(nav-bar-functionality): wire up hamburger button with panel

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

image

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/common/components/expand-collapse-left-nav-hamburger-button.test.tsx [hunk 1]
```diff
@@ -9,15 +9,27 @@ import * as React from 'react';
 
 describe('AssessmentLeftNavHamburgerButton', () => {
     it('renders per snapshot', () => {
-        const wrapper = shallow(<AssessmentLeftNavHamburgerButton isLeftNavOpen={false} />);
+        const wrapper = shallow(
+            <AssessmentLeftNavHamburgerButton
+                isSideNavOpen={false}
+                setSideNavOpen={null}
+                isLeftNavOpen={false}
```
Comment (lisli1) [inline src/tests/unit/tests/common/components/expand-collapse-left-nav-hamburger-button.test.tsx:16]: This looks like it was renamed to isSideNavOpen?

File: src/tests/unit/tests/common/components/expand-collapse-left-nav-hamburger-button.test.tsx [hunk 2]
```diff
@@ -9,15 +9,27 @@ import * as React from 'react';
...
 });
 
 describe('FastPassLeftNavHamburgerButton', () => {
     it('renders per snapshot', () => {
-        const wrapper = shallow(<FastPassLeftNavHamburgerButton isLeftNavOpen={false} />);
+        const wrapper = shallow(
+            <FastPassLeftNavHamburgerButton
+                isSideNavOpen={false}
+                setSideNavOpen={null}
+                isLeftNavOpen={false}
```
Comment (lisli1) [inline src/tests/unit/tests/common/components/expand-collapse-left-nav-hamburger-button.test.tsx:30]: Same as above

File: src/DetailsView/components/details-view-content-with-local-state.tsx [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
 
 export type DetailsViewContentWithLocalStateProps = DetailsViewContainerProps;
+export type DetailsViewContentState = {
+    isSideNavOpen: boolean;
+};
 
-export const DetailsViewContentWithLocalState = NamedFC<DetailsViewContentWithLocalStateProps>(
-    'DetailsViewContentWithLocalState',
-    props => {
-        const [isSideNavOpen, setSideNavOpen] = React.useState(false);
```
Comment (waabid) [inline src/DetailsView/components/details-view-content-with-local-state.tsx:13]: How come this didn't work?

File: src/common/components/left-nav-hamburger-button.tsx [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
 // Licensed under the MIT License.
-
 import { NamedFC } from 'common/react/named-fc';
 import { IconButton } from 'office-ui-fabric-react';
 import * as React from 'react';
+
 import * as styles from './left-nav-hamburger-button.scss';
 
-type LeftNavHamburgerButtonProps = {
+export type LeftNavHamburgerButtonProps = {
```
Comment (waabid) [inline src/common/components/left-nav-hamburger-button.tsx:9]: This can probably extend the `ExpandCollpaseLeftNavButtonProps`

**Pull request merged**